client/src/App.jsx

```diff
@@ -296,11 +300,22 @@ function App() {
                   loginUser={loginUser}
                 >
                   <ManagedUsersLoader setManagedUsers={setManagedUsers}>
-                    <EmployerPage
-                      getUserById={getUserById}
-                      managedUsers={managedUsers}
-                      setSnackBarMessage={setSnackBarMessage}
-                    />
+                    <ManagedJobLeadsLoader
+                      setManagedJobLeads={setManagedJobLeads}
+                    >
+                      <ClientsLoader setClients={setManagedClients}>
+                        <EmployersLoader setEmployers={setEmployers}>
+                          <EmployerPage
```

Comment:
Hey @DanielDervishi, very good point thank for calling this out. I have fixed the fetching model such that we only fetch when the user decides to initiate such placement to load the assoicated job leads and clients. I made it so that for employer page timeline placement, you only get job leads that are asssoicated to that employer. In the Routing V2 ticket I will work towards a smarter pagination model in the dropdown combobox but for now I hope the current fix should be fine. :)